src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -116,28 +116,67 @@ private function parseDefinitions(\DOMDocument $xml, $file)
         }
 
         foreach ($services as $service) {
-            if (null !== $definition = $this->parseDefinition($service, $file)) {
+            if (null !== $definition = $this->parseDefinition($service, $file, $this->getServiceDefaults($xml))) {
                 $this->container->setDefinition((string) $service->getAttribute('id'), $definition);
             }
         }
     }
 
     /**
+     * Get service defaults
+     *
+     * @param \DOMDocument $xml
+     *
+     * @return array
+     */
+    private function getServiceDefaults(\DOMDocument $xml)
+    {
+        $xpath = new \DOMXPath($xml);
+        $xpath->registerNamespace('container', self::NS);
+
+        $defaults = array();
+
+        if (null === $defaultsNode = $xpath->query('//container:services/container:defaults')->item(0)) {
+            return $defaults;
+        }
+
+        if ($defaultsNode->hasAttribute('public')) {
+            $defaults['public'] = XmlUtils::phpize($defaultsNode->getAttribute('public'));
+        }
+
+        if ($defaultsNode->hasAttribute('autowire')) {
+            $defaults['autowire'] = XmlUtils::phpize($defaultsNode->getAttribute('autowire'));
+        }
+
+        $defaults['tags'] = $this->getChildren($defaultsNode, 'tag');
+
+        $defaults['autowire_methods'] = [];
+        foreach ($this->getChildren($defaultsNode, 'autowire') as $type) {
+            $defaults['autowire_methods'][] = $type->textContent;
```

Comment:
this misses the validation of the attribute and child element not being used together (better error message if given here)